### PR TITLE
replication: support update replication mode online 

### PR DESCRIPTION
**Pull request by disksing**
### What problem does this PR solve? 
Fix: https://github.com/pingcap/pd/issues/2317

### What is changed and how it works?
- persist replication mode config
- notify replicationModeManager when config is updated
- handle special case when mode changed from majority to dr_auto_sync
- add API to update config
- add pd-ctl command to update config
- add test

### Check List 

Tests 
 - Unit test
 - Manual test (add detailed scripts or steps below)

Code changes
 - Has HTTP API interfaces change (Don't forget to [update API document](https://github.com/pingcap/pd/blob/master/docs/development.md#updating-api-documentation))

### CODE REVIEW

File: tools/pd-ctl/pdctl/command/config_command.go [hunk 1]
```diff
@@ -249,6 +268,15 @@ func showClusterVersionCommandFunc(cmd *cobra.Command, args []string) {
 	cmd.Println(r)
 }
 
+func showReplicationModeCommandFunc(cmd *cobra.Command, args []string) {
+	r, err := doRequest(cmd, replicationModePrefix, http.MethodGet)
+	if err != nil {
+		cmd.Printf("Failed to get replication mode config:%s\n", err)
```
Comment (rleungx) [inline tools/pd-ctl/pdctl/command/config_command.go:274]:
```suggestion
		cmd.Printf("Failed to get replication mode config: %s\n", err)
```

File: server/api/config.go [hunk 2]
```diff
@@ -307,3 +307,34 @@ func (h *confHandler) SetClusterVersion(w http.ResponseWriter, r *http.Request)
...
+}
+
+// @Tags config
+// @Summary Set replication mode config.
+// @Accept json
+// @Param body body object string "json params"
+// @Produce json
+// @Success 200 {string} string
+// @Failure 500 {string} string "PD server failed to proceed the request."
+// @Failure 503 {string} string "PD server has no leader."
```
Comment (rleungx) [inline server/api/config.go:327]: the code will not return 503 error?

File: server/api/router.go [hunk 3]
```diff
@@ -86,6 +86,8 @@ func createRouter(ctx context.Context, prefix string, svr *server.Server) *mux.R
 	apiRouter.HandleFunc("/config/label-property", confHandler.SetLabelProperty).Methods("POST")
 	apiRouter.HandleFunc("/config/cluster-version", confHandler.GetClusterVersion).Methods("GET")
 	apiRouter.HandleFunc("/config/cluster-version", confHandler.SetClusterVersion).Methods("POST")
+	apiRouter.HandleFunc("/config/replication-mode", confHandler.GetReplicationMode).Methods("GET")
+	apiRouter.HandleFunc("/config/replication-mode", confHandler.SetReplicationMode).Methods("POST")
```
Comment (rleungx) [inline server/api/router.go:90]: Do we also need to support changing it through /config API?

Comment (disksing) [reply server/api/router.go:90]: I guess it is ok to not support. Just like `cluster-version`, `label-property`, etc.

Comment (rleungx) [reply server/api/router.go:90]: Does this config need SQL support? If it needs, I think it's better to support changing through /config. For `cluster-version` and `label-property`, PR #2354 is trying to support it.

**Pull request merged**